File: pat1035.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;

int main() {

	int N;
	cin >> N;
	string IDs[1000], passwords[1000];
	vector<int> modified_index;
	for (int i = 0; i < N; i++) {
		string ID, password;
		cin >> ID >> password;
		IDs[i] = ID;
		passwords[i] = password;

	}
	int modified_cnt = 0;
	for (int i = 0; i < N; i++) {
		string new_password = "";
		for (int j = 0; j < passwords[i].length(); j++) {
				if (passwords[i][j] == '1') {
					new_password += '@';
				}
				else if (passwords[i][j] == '0'){
					new_password += '%';
				}
				else if (passwords[i][j] == 'l') {
					new_password += 'L';
				}
				else if (passwords[i][j] == 'O') {
					new_password += 'o';
				}
				else {
					new_password += passwords[i][j];
				}
				
		}
		if (new_password.compare(passwords[i]) != 0) {
			modified_cnt++;
			passwords[i] = new_password;
			modified_index.push_back(i);
		}
	}

	if (modified_cnt == 0) {
		if (N == 1) {
			cout << "There is " << N << " account and no account is modified";
		}
		else {
			cout << "There are " << N << " accounts and no account is modified";
		}
	}
	else {
		cout << modified_cnt << endl;
		for (int i = 0; i < modified_cnt; i++) {
			int index = modified_index[i];
			cout << IDs[index] << ' ' << passwords[index] << endl;
		}
	}


	return 0;
}

File: pat1035.py
'''
    思路：可以直接调用string的translate解题
    注意：密码没改变时N=1和N>1输出语句不同
'''

N = int(input())
accounts = []
for x in range(N):
    ID,password = input().split()
    account = {}
    account['ID'] = ID
    account['password'] = password
    accounts.append(account)
origin_string = '10lO'
new_string = '@%Lo'
modified_cnt = 0
modified_account = []
for account in accounts:
    str_trans = str.maketrans(origin_string,new_string)
    new_password = account['password'].translate(str_trans)
    if new_password != account['password']:
        modified_cnt += 1
        account['password'] = new_password
        modified_account.append(account['ID'])

if modified_cnt == 0:
    if(N == 1):
        print('There is ' + str(N) + ' account and no account is modified')
    else:
        print('There are ' + str(N) + ' accounts and no account is modified')
else:
    print(modified_cnt)
    for account in accounts:
        if account['ID'] in modified_account:
            print(account['ID'] + ' ' + account['password'])
